examples/nrf52/microbit/esp8266/mqtt/src/main.rs

```diff
@@ -0,0 +1,231 @@
+            ssid: WIFI_SSID.trim_end(),
+            password: WIFI_PSK.trim_end(),
+        })
+        .await
+        .expect("Error joining WiFi network");
+
+    static NETWORK: TcpStackState<Esp8266Controller<'static, TX>> = TcpStackState::new();
+    let mut network = NETWORK.initialize(network);
+
+
+
+    let ips = DNS.resolve(HOST).await.expect("unable to resolve host");
+
+
+    defmt::info!("Creating sockets");
+    let mut socket_publish = Socket::new(network.clone(), network.open().await.unwrap());
+
+    socket_publish
+        .connect(
+            IpProtocol::Tcp,
+            SocketAddress::new(ips[0], PORT.parse::<u16>().unwrap()),
+        )
+        .await
+        .expect("Error creating connection");
+
+    let mut socket_receiver = Socket::new(network.clone(), network.open().await.unwrap());
+
+    socket_receiver
+        .connect(
+            IpProtocol::Tcp,
+            SocketAddress::new(ips[0], PORT.parse::<u16>().unwrap()),
+        )
+        .await
+        .expect("Error creating connection");
+
+    static RECEIVER: ActorContext<Receiver> = ActorContext::new();
+    RECEIVER.mount(
+        spawner,
+        Receiver::new(board.display, DrogueNetwork::new(socket_receiver)),
+    );
+
+
+    let mut config = ClientConfig::new(MQTTv5, CountingRng(0));
+    config.add_qos(QualityOfService::QoS0);
+    config.keep_alive = u16::MAX;
+    let mut recv_buffer = [0; 100];
+    let mut write_buffer = [0; 100];
+
+    let mut client = MqttClient::<_, 20, CountingRng>::new(
+        DrogueNetwork::new(socket_publish),
+        &mut write_buffer,
+        100,
+        &mut recv_buffer,
+        100,
+        config,
+    );
+    defmt::info!("[PUBLISHER] Connecting to broker");
+    client.connect_to_broker().await.unwrap();
+    let mut button = board.btn_a;
+    loop {
+        defmt::info!("[PUBLISHER] Press 'A' button to send data");
+        button.wait_pressed().await;
+        defmt::info!("[PUBLISHER] sending message");
+        client.send_message(TOPIC, "Hello world!").await.unwrap();
+        defmt::info!("[PUBLISHER] message sent");
+    }
+}
+
+static DNS: StaticDnsResolver<'static, 2> = StaticDnsResolver::new(&[
+    DnsEntry::new("localhost", IpAddress::new_v4(127, 0, 0, 1)),
+    DnsEntry::new(
+        "broker.emqx.io",
+        IpAddress::new_v4(54, 184, 34, 50),
+    )
+]);
+
+#[embassy::task]
+pub async fn wifi(mut modem: Esp8266Modem<'static, RX, ENABLE, RESET>) {
+    modem.run().await;
+}
+
+
+pub struct Receiver {
+    display: LedMatrix,
+    socket: Option<DrogueNetwork<Handle<'static, Esp8266Controller<'static, TX>>>>,
+}
+
+impl Receiver {
+    pub fn new(display: LedMatrix, socket: DrogueNetwork<Handle<'static, Esp8266Controller<'static, TX>>>) -> Self {
+        Self {
+            display,
+            socket: Some(socket),
+        }
+    }
+}
+
+#[derive(Clone)]
+pub enum ReceiverMessage {
+    Toggle,
+}
+
+#[actor]
+impl Actor for Receiver {
+    type Message<'m> = ReceiverMessage;
```

Comment:
Afaict, the receiver is not really receiving anything. Maybe just create a 'async receiver()' function and spawn that?

Signature like this:
```
async fn receiver(display: LedMatrix, socket: DrogueNetwork<Handle<'static, Esp8266Controller<'static, TX>>>)  {}
```
And then spawn as any other task.